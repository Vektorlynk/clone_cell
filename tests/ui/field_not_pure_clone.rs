use clone_cell::{cell::Cell, clone::PureClone};

#[derive(PureClone)]
struct Foo {
    x: Cell<i32>,
}

fn main() {}
