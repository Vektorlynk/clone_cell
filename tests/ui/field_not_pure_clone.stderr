error[E0277]: the trait bound `Foo: PureClone` is not satisfied
  --> $DIR/field_not_pure_clone.rs:8:8
   |
8  |     f: Foo,
   |        ^^^ the trait `PureClone` is not implemented for `Foo`
   |
note: required by `pure_clone`
  --> $DIR/clone.rs:38:5
   |
38 |     fn pure_clone(&self) -> Self {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
